public/app/controllers/versionbumper-overview.controller.js

```diff
@@ -1,30 +1,168 @@
 "use strict";
 
 angular
-	.module("acpaasportalversionbumper_0.0.1.controllers")
+	.module("acpaasportalversionbumper_0.0.9.controllers")
 	.controller("acpaasportalversionbumperOverviewController", [
 		"$scope",
 		"$timeout",
+		"$q",
 		"acpaasportalversionbumperFactory",
+		"acpaasportalversionbumperConfig",
 		"LabelService",
-		"NotificationService",
+		"DialogService",
 
 		function(
 			$scope,
 			$timeout,
+			$q,
 			acpaasportalversionbumperFactory,
+			acpaasportalversionbumperConfig,
 			LabelService,
-			NotificationService
+			DialogService
 		) {
+			$scope.filters = {
+				type: "",
+				product: "",
+				api: "",
+				searchQuery: "",
+			};
+			$scope.types = [];
+			$scope.loading = {
+				types: false,
+				table: false,
+				search: false,
+			};
 			$scope.products = [];
+			$scope.apis = [];
+			$scope.tableData = [];
+			$scope.searchFieldOptions = {
+				debounce: 150,
+			};
+			$scope.pagination = {
+				currentPage: 0,
+				itemsPerPage: 10,
+				async: false,
+				itemsLength: 0,
+			};
+			$scope.tableConfig = {
+				pagination: $scope.pagination,
+				selects: [],
+				searchField: {
+					enabled: false,
+				},
+				columns: [{
+					columnName: LabelService.getString("Name"),
+					key: "meta.label",
+					sortable: true,
+				}, {
+					columnName: LabelService.getString("Description"),
+					key: "meta.description",
+					sortable: true,
+				}, {
+					columnName: LabelService.getString("Author"),
+					key: "meta.lastEditor.user"
+				}, {
+					columnName: LabelService.getString("Last edit"),
+					key: "meta.lastModified",
+					mode: "timeAgo",
+					defaultSort: true,
+					sortable: true,
+				}, {
+					columnName: LabelService.getString("Actions"),
+					template: "<a ng-click=\"$emit('bumpVersion', i)\">" + LabelService.getString("Bump") + "</a>",
+				}],
+			};
 
 			function init() {
-				acpaasportalversionbumperFactory.fetchProducts()
-					.then(function(products) {
-						$scope.products = products;
+				fetchTypes();
+			}
+
+			function fetchTypes() {
```

Comment:
Fetching van data die altijd ingeladen wordt bij het inladen van de pagina doen we via de router resolves.
Er zijn aan beide methodes een voordeel verbonden maar i.f.v. concistentie lijkt het me beter om hier een resolve te gebruiken.